cpp/open3d/visualization/rendering/Scene.h

```diff
@@ -41,83 +42,110 @@ class AxisAlignedBoundingBox;
                                std::int32_t y,
                                std::uint32_t w,
                                std::uint32_t h) = 0;
+
     virtual View* GetView(const ViewHandle& view_id) const = 0;
     virtual void SetViewActive(const ViewHandle& view_id, bool is_active) = 0;
     virtual void RemoveView(const ViewHandle& view_id) = 0;
 
-    // 'All defaults' way:
-    // * Will use geometry name as entity name
-    // * Will use apropriate default material for rendering
-    // * For geometries with textures, will try load and assign a texture
-    virtual GeometryHandle AddGeometry(
-            const geometry::Geometry3D& geometry) = 0;
-    // Will use geometry name as entity name
-    virtual GeometryHandle AddGeometry(
-            const geometry::Geometry3D& geometry,
-            const MaterialInstanceHandle& material_id) = 0;
-    virtual GeometryHandle AddGeometry(
-            const geometry::Geometry3D& geometry,
-            const MaterialInstanceHandle& material_id,
-            const std::string& name) = 0;
-    virtual void AssignMaterial(const GeometryHandle& geometry_id,
-                                const MaterialInstanceHandle& material_id) = 0;
-    virtual MaterialInstanceHandle GetMaterial(
-            const GeometryHandle& geometry_id) const = 0;
-    virtual void SetGeometryShadows(const GeometryHandle& geometry_id,
-                                    bool casts_shadows,
-                                    bool receives_shadows) = 0;
-    virtual std::vector<GeometryHandle> FindGeometryByName(
-            const std::string& name) = 0;
-    virtual void RemoveGeometry(const GeometryHandle& geometry_id) = 0;
-
-    virtual LightHandle AddLight(const LightDescription& descr) = 0;
-    // TODO: If possible, add getters
-    virtual void SetLightIntensity(const LightHandle& id, float intensity) = 0;
-    virtual void SetLightColor(const LightHandle& id,
-                               const Eigen::Vector3f& color) = 0;
-    virtual Eigen::Vector3f GetLightDirection(const LightHandle& id) const = 0;
-    virtual void SetLightDirection(const LightHandle& id,
-                                   const Eigen::Vector3f& dir) = 0;
-    virtual void SetLightPosition(const LightHandle& id,
-                                  const Eigen::Vector3f& pos) = 0;
-    virtual void SetLightFalloff(const LightHandle& id, float falloff) = 0;
-    virtual void RemoveLight(const LightHandle& id) = 0;
-
-    // Passing empty id disables indirect lightning
-    virtual void SetIndirectLight(const IndirectLightHandle& id) = 0;
+    // Camera
+    virtual void AddCamera(const std::string& camera_name,
+                           std::shared_ptr<Camera> cam) = 0;
+    virtual void RemoveCamera(const std::string& camera_name) = 0;
+    virtual void SetActiveCamera(const std::string& camera_name) = 0;
+
+    // Scene geometry
+    virtual bool AddGeometry(const std::string& object_name,
```

Comment:
I would prefer SetGeometry() which would replace an existing geometry with the same name. Of course remove+add is the same but set seems more user friendly.